complete/src/main/resources/static/app.js

```diff
@@ -18,100 +18,134 @@ function refreshTimeTable() {
         var theadByRoom = $("<thead>").appendTo(timeTableByRoom);
         var headerRowByRoom = $("<tr>").appendTo(theadByRoom);
         headerRowByRoom.append($("<th>Timeslot</th>"));
-        $.each(timeTable.roomList, function (index, room) {
-            headerRowByRoom.append($("<th>"
-                    + "<span>" + room.name + "</span>"
-                    + "<button id=\"deleteRoomButton-" + room.id + "\" type=\"button\" class=\"ml-2 mb-1 btn btn-light btn-sm p-1\">"
-                    + "<small class=\"fas fa-trash\"></small>"
-                    + "</button>"
-                    + "</th>"));
-            $("#deleteRoomButton-" + room.id).click(function() {
+        $.each(timeTable.roomList, (index, room) => {
+            const id = uuid();
+            headerRowByRoom.append($(template({ room }, props => `
+                <th>
+                    <span>${props.room.name}</span>
+                    <button id="${id}" type="button" class="ml-2 mb-1 btn btn-light btn-sm p-1">
+                        <small class="fas fa-trash"></small>
+                    </button>
+                </th>`
+            )));
+            $(`#${id}`).click(function() {
                 deleteRoom(room);
             });
         });
         var theadByTeacher = $("<thead>").appendTo(timeTableByTeacher);
         var headerRowByTeacher = $("<tr>").appendTo(theadByTeacher);
         headerRowByTeacher.append($("<th>Timeslot</th>"));
         const teacherList = [...new Set(timeTable.lessonList.map(lesson => lesson.teacher))];
-        $.each(teacherList, function (index, teacher) {
-            headerRowByTeacher.append($("<th>"
-                    + "<span>" + teacher + "</span>"
-                    + "</th>"));
+        $.each(teacherList, (index, teacher) => {
+            headerRowByTeacher.append($(template({ teacher }, props => `
+                <th>
+                    <span>${props.teacher}</span>
+                </th>`
+            )));
```

Comment:
This template pattern is rather verbose.
A StackOverflow question seems to indicate that something like this would also work:

`headerRowByTeacher.html("").append($("<span></span>").text(teacher))`

TODO that clearly missed the `<th>` creation.
If we can get that style working, the XSS safe text would be simpler to read, no?

Reply:
New style:
```
            headerRowByTeacher
            .append($("<th>")
                .append($("<span />").text(teacher))
            .append("</th>"));
```
Notes:
This style is harder to read (for me) and more prone to bugs than the previous style. For instance, most of the time, you need to be sure to create a new jquery element within the append, otherwise very subtle bugs will occur in the displayed html. The only advantage is the ability to add click handlers next to buttons.